apollo-federation/src/query_plan/operation.rs

```diff
@@ -2222,8 +2283,179 @@ impl SelectionSet {
         Ok(conditions)
     }
 
+    fn make_selection(
+        schema: &ValidFederationSchema,
+        parent_type: &CompositeTypeDefinitionPosition,
+        updates: Vec<Selection>,
```

Comment:
This should take a slice. This is partially because I need this to take a slice for the `add_at_path` work, but also there is no need for this function to take an owned `Vec`.
```suggestion
        updates: &[Selection],
```